internal/subscription/core.go

```diff
@@ -110,6 +112,44 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 	return c.repo.Save(ctx, m)
 }
 
+//UpdateSubscription Updates an exisiting subscription at the provided update path
+func (c *Core) UpdateSubscription(ctx context.Context, uModel *Model, paths []string) (*Model, error) {
+	subscriptionOperationCount.WithLabelValues(env, "UpdateSubscription").Inc()
+
+	startTime := time.Now()
+	defer func() {
+		subscriptionOperationTimeTaken.WithLabelValues(env, "UpdateSubscription").Observe(time.Now().Sub(startTime).Seconds())
+	}()
+
+	if ok, err := c.projectCore.ExistsWithID(ctx, uModel.ExtractedSubscriptionProjectID); !ok {
+		if err != nil {
+			return nil, err
+		}
+		logger.Ctx(ctx).Errorw("subscription project not found", "name", uModel.ExtractedSubscriptionProjectID)
+		return nil, merror.New(merror.NotFound, "project not found")
+	}
+
+	if ok, err := c.Exists(ctx, uModel.Key()); !ok {
+		if err != nil {
+			return nil, err
+		}
+		logger.Ctx(ctx).Errorw("subscription not found", "name", uModel.Name)
+		return nil, merror.New(merror.NotFound, "subscription not found")
+	}
+
+	m, err := c.Get(ctx, uModel.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	err = patch.StructPatch(ctx, uModel, m, paths)
```

Comment:
should every model call patch pkg ?
can we abstract it out a bit

something like m.Patch(ctx, uModel, paths) 

and baseModel calling pkg and handling it, this way in future if we change the pkg we need to make changes at only one place

Reply:
Let's say I define the method `Patch` in the IModel interface. And this will be implemented by the `BaseModel`. The `BaseModel` would implement the method.
`func (b *BaseModel) Patch(...){}`
`BaseModel` would be embedded inside evry other model. This would promote the implementation into each and every model. However, when invoked, this method would still receive the instance of BaseModel and not of the actual model. So the only other way to do this would be to implement it at every model level.. which is what we don't want.

Reply:
> this would be to implement it at every model level.. which is what we don't want.
Dont see issues here as long as we are not exposing it via Service/Core layer thought i get the point that BaseModel wont work, we need to solve it via composition to avoid duplication later

Reply:
Will resolve conversation?

Reply:
@sunnyagg So the current implementation works for now?